test/index.test.ts

```diff
@@ -143,10 +143,10 @@ describe('FiatConnect SDK', () => {
       fetchMock.mockRejectOnce(new Error('fake error message'))
       const response = await client.getClockDiffApprox()
 
-      expect(response.ok).toBeFalsy()
-      expect(response.val).toMatchObject({
-        error: 'fake error message',
-      })
+      expect(response.isOk).toBeFalsy()
+      expect(response.unwrap.bind(response)).toThrow(
```

Comment:
small note: this isn't how we will normally consume errors. the use of `unwrap` at all is a hack to work around the type checker.

previously we were able to access _either_ the resolved value or error through `result.val`, so the type checker would never complain at us when accessing `result.val`. now, the resolved value is available at `result.value` and the error at `result.error`. normally, checking `response.isOk` or `response.isErr` acts as a type guard so the compiler will allow us to access `result.value`/`result.error` after the check... these `expect` calls do not act as a type guard however, so we use `unwrap` here which either returns the result or throws the error, depending.

Reply:
Another potential hack for this is doing
```js
expect(response.isErr && response.error).toEqual(..)
```
or 
```js
expect(response.isOk && response.value).toEqual(...)
```
but don't feel strongly either way.